bson/src/main/org/bson/codecs/configuration/mapper/ClassModel.java

```diff
@@ -34,6 +35,10 @@
     private final Map<String, FieldModel> fields = new TreeMap<String, FieldModel>();
     private final CodecRegistry registry;
     private final TypeResolver resolver;
+    private final MemberResolver memberResolver;
+    private final WeightedValue<String> collectionName = new WeightedValue<String>();
```

Comment:
I'm of two minds about having this in here. Strictly speaking, this is just codec and should know anything about a collection. For instance, what if this class is intended for use in more than one collection?